doc/nrf/protocols/matter/end_product/security.rst

```diff
@@ -1,16 +1,108 @@
+Such component is essential for IoT devices and it allows to implement secure communication between devices within the Matter ecosystems.
+
+Depending on the hardware capabilities of the used platform, the Trusted Storage can utilize one of the following backends:
+
+* `ARM TrustZone`_ technology and hardware accelerated firmware isolation
+*  :ref:`trusted_storage_readme` |NCS| software library
+
+Currently all :ref:`Matter samples <matter_samples>` use Trusted Storage library as a trusted storage backend for all supported platforms.
+
+Trusted Storage library provides confidentiality, integrity, and authenticity for the assets it operates on.
+It handles sensitive data in two contexts:
+
+* volatile - before data is forwarded to the non-volatile storage and after it is retrieved from the non-volatile storage
+* non-volatile - while data is being written to the non-volatile memory
+
+In the case of volatile context, the Trusted Storage library leverages the Authenticated Encryption with Associated Data (AEAD) encryption backend (:kconfig:option:`CONFIG_TRUSTED_STORAGE_BACKEND_AEAD`).
+AEAD can be configured with the set of Kconfig options described in the :ref:`Trusted Storage configuration <trusted_storage_configuration>` section.
+
+An important setting, that depends on the hardware platform in use, is the way of generating the AEAD keys.
+The recommended and the most secure option is to use :ref:`Hardware Unique Key (HUK) library <lib_hw_unique_key>` for that purpose.
+HUK support is automatically enabled with :kconfig:option:`CONFIG_TRUSTED_STORAGE_BACKEND_AEAD_KEY_DERIVE_FROM_HUK` Kconfig option for compatible platforms.
+Note that, due to hardware limitations, this library is not available for nRF52840 family.
+In that case, the AEAD keys can be derived with SHA-256 hash (:kconfig:option:`CONFIG_TRUSTED_STORAGE_BACKEND_AEAD_KEY_HASH_UID`) but this approach is less secure than using HUK for key derivation as it will only provide integrity of sensitive material.
+It is also possible to implement a custom AEAD keys generation method when the :kconfig:option:`CONFIG_TRUSTED_STORAGE_BACKEND_AEAD_KEY_CUSTOM` Kconfig option is selected.
+
+You can find an overview of Trusted Storage layer configuration supported for each |NCS| Matter-enabled platform in `Matter platforms security support`_ section.
+
+Matter platforms security support
+*********************************
+
+The following table summarizes current security configuration and features supported for Matter-enabled hardware platforms in |NCS|.
+
+.. list-table:: Matter platforms security support
+   :widths: auto
+   :header-rows: 1
+
+   * - Platform
+     - Networking backend
+     - Cryptography API
+     - Cryptography backend
+     - ARM TrustZone support
+     - Trusted Storage backend
+     - Trusted Storage AEAD key
+   * - nRF52840
```

Comment:
For all of the platforms, should they be SoC, SiP, or DK?

Reply:
SoC, I'd say.